internal/state/store.go

```diff
@@ -931,3 +982,11 @@ func int64ToBytes(i int64) []byte {
 	n := binary.PutVarint(buf, i)
 	return buf[:n]
 }
+
+// addTimeSample returns a function that, when called, adds an observation to m.
+// The observation added to m is the number of seconds elapsed since addTimeSample
+// was initially called. addTimeSample is meant to be called in a defer to calculate
+// the amount of time a function takes to complete.
+func addTimeSample(m metrics.Histogram, start time.Time) func() {
+	return func() { m.Observe(time.Since(start).Seconds()) }
```

Comment:
Yes, it address. Please disregard for now. Changing this would require many other changes and since `Seconds()` return a `flota64` it's all good. Thanks !